packages/server/src/utils/databases/mongo.ts

```diff
@@ -30,15 +30,54 @@ export const buildMongo = async (mongo: MongoNested) => {
 		mounts,
 	} = mongo;
 
-	const defaultMongoEnv = `MONGO_INITDB_ROOT_USERNAME=${databaseUser}\nMONGO_INITDB_ROOT_PASSWORD=${databasePassword}${
-		env ? `\n${env}` : ""
-	}`;
+	const initReplicaSet = `
+#!/bin/bash
+mongod --port 27017 --replSet rs0 --bind_ip_all &
+MONGOD_PID=$!
+
+# Wait for MongoDB to be ready
+while ! mongosh --eval "db.adminCommand('ping')" > /dev/null 2>&1; do
+  sleep 2
+done
+
+# Check if replica set is already initialized
+REPLICA_STATUS=$(mongosh --quiet --eval "rs.status().ok || 0")
+
+if [ "$REPLICA_STATUS" != "1" ]; then
+  echo "Initializing replica set..."
+  mongosh --eval '
+    rs.initiate({
+      _id: "rs0",
+      members: [{ _id: 0, host: "localhost:27017", priority: 1 }]
+    });
+
+    // Wait for the replica set to initialize
+    while (!rs.isMaster().ismaster) {
+      sleep(1000);
+    }
+
+    // Create root user after replica set is initialized and we are primary
+    db.getSiblingDB("admin").createUser({
+      user: "${databaseUser}",
+      pwd: "${databasePassword}",
+      roles: ["root"]
+    });
+  '
+else
+  echo "Replica set already initialized."
+fi
+
+wait $MONGOD_PID`;
+
+	const defaultMongoEnv = `MONGO_INITDB_DATABASE=admin\n${env ? `${env}` : ""}`;
+
```

Comment:
I am not an expert in mongo but it seems to work, however I do not want to change the behavior of the way mongo works because of certain libraries, I think that to manage the current way and the way of replicas, we could manage some checkbox in the interface at the time of creating the database and so we will use either this configuration that you implemented and then by default will be what currently has dokploy.

Reply:
There is no downside to using a single node replica set, you are just making it harder for people that want to use mongodb and dokploy.

Reply:
The sole difference between a single node replica set and a standalone database is that a replica set takes slightly more disk space because of the logs mongo uses to sync between the replica set nodes, I'll add a toggle for it. Should it be on by default or have the default be the original standalone one @Siumauricio

Reply:
@thewilloftheshadow 
Thanks for the clarification, by default I would like the replica nodes to be disabled by default, and the default value to be what we currently have in dokploy.

Reply:
why not let people choose between both options ?

Reply:
> why not let people choose between both options ?

What two options are you talking about? Replica sets on or of? Or single/multi node replica sets?

Reply:
> > why not let people choose between both options ?
> 
> What two options are you talking about? Replica sets on or of? Or single/multi node replica sets?

yes